### PR TITLE
SkipLocalsInitAttribute on classes

### PR DESCRIPTION
**Pull request by t-camaia**
This change enables applying SkipLocalsInitAttribute directly on classes. It can be put on a global level class or on a nested one.

### CODE REVIEW

File: src/Compilers/Core/Portable/Symbols/Attributes/CommonTypeWellKnownAttributeData.cs [hunk 1]
```diff
@@ -246,5 +246,23 @@ public bool HasExcludeFromCodeCoverageAttribute
         }
 
         #endregion
+
+        #region SkipLocalsInitAttribute
+        private bool _hasSkipLocalsInitAttribute;
```
Comment (AlekseyTs) [inline src/Compilers/Core/Portable/Symbols/Attributes/CommonTypeWellKnownAttributeData.cs]: >_hasSkipLocalsInitAttribute [](start = 21, length = 27)

Please move this to C# specific structure and do the same for other symbols

File: src/Compilers/CSharp/Portable/Symbols/Source/SourceNamedTypeSymbol.cs [hunk 2]
```diff
@@ -944,6 +948,21 @@ public sealed override bool IsSerializable
...
 
+        public sealed override bool AreLocalsZeroed
+        {
+            get
+            {
+                var data = this.GetDecodedWellKnownAttributeData();
+
+                if (data != null)
+                {
+                    return !data.HasSkipLocalsInitAttribute;
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Portable/Symbols/Source/SourceNamedTypeSymbol.cs]: >return !data.HasSkipLocalsInitAttribute; [](start = 20, length = 40)

This looks wrong.

**Comment by AlekseyTs** [comment]
Done with review pass (iteration 2)

File: src/Compilers/CSharp/Portable/Symbols/Source/SourceNamedTypeSymbol.cs (hunk 2 above)
Comment (t-camaia) [reply src/Compilers/CSharp/Portable/Symbols/Source/SourceNamedTypeSymbol.cs]: Yes, that's wrong. Just forgot to update this. I'll fix it and add a test.

File: src/Compilers/Core/Portable/Symbols/Attributes/CommonTypeWellKnownAttributeData.cs (hunk 1 above)
Comment (t-camaia) [reply src/Compilers/Core/Portable/Symbols/Attributes/CommonTypeWellKnownAttributeData.cs]: Done.

**Comment by t-camaia** [comment]
@AlekseyTs Just updated it with your suggested changes. Also, I implemented @gafter's suggestion of adding a test helper method to look for '.locals init' instead of always verifying the whole IL.

File: src/Compilers/CSharp/Portable/Symbols/Attributes/WellKnownAttributeData/MethodWellKnownAttributeData.cs [hunk 3]
```diff
@@ -0,0 +1,30 @@
+﻿using System;
```
Comment (cston) [inline src/Compilers/CSharp/Portable/Symbols/Attributes/WellKnownAttributeData/MethodWellKnownAttributeData.cs]: Please add the copyright header.

File: src/Compilers/CSharp/Portable/Symbols/Attributes/WellKnownAttributeData/PropertyWellKnownAttributeData.cs [hunk 4]
```diff
@@ -0,0 +1,30 @@
+﻿using System;
```
Comment (cston) [inline src/Compilers/CSharp/Portable/Symbols/Attributes/WellKnownAttributeData/PropertyWellKnownAttributeData.cs]: Copyright header.

File: src/Compilers/CSharp/Portable/Symbols/Attributes/WellKnownAttributeData/MethodWellKnownAttributeData.cs [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+using System.Collections.Generic;
+using System.Text;
+
+namespace Microsoft.CodeAnalysis.CSharp.Symbols
+{
+    /// <summary>
+    /// Information decoded from well-known custom attributes applied on a method.
+    /// </summary>
+    class MethodWellKnownAttributeData : CommonMethodWellKnownAttributeData
```
Comment (cston) [inline src/Compilers/CSharp/Portable/Symbols/Attributes/WellKnownAttributeData/MethodWellKnownAttributeData.cs]: `internal sealed`

File: src/Compilers/CSharp/Portable/Symbols/Attributes/WellKnownAttributeData/MethodWellKnownAttributeData.cs (hunk 3 above)
Comment (t-camaia) [reply src/Compilers/CSharp/Portable/Symbols/Attributes/WellKnownAttributeData/MethodWellKnownAttributeData.cs]: Done.

File: src/Compilers/CSharp/Portable/Symbols/Attributes/WellKnownAttributeData/PropertyWellKnownAttributeData.cs (hunk 4 above)
Comment (t-camaia) [reply src/Compilers/CSharp/Portable/Symbols/Attributes/WellKnownAttributeData/PropertyWellKnownAttributeData.cs]: Done.

File: src/Compilers/CSharp/Portable/Symbols/Attributes/WellKnownAttributeData/MethodWellKnownAttributeData.cs (hunk 5 above)
Comment (t-camaia) [reply src/Compilers/CSharp/Portable/Symbols/Attributes/WellKnownAttributeData/MethodWellKnownAttributeData.cs]: My bad. Fixed it.

**Comment by t-camaia** [comment]
test windows_debug_unit64_prtest

File: src/Compilers/CSharp/Portable/Symbols/Attributes/WellKnownAttributeData/TypeWellKnownAttributeData.cs [hunk 6]
```diff
@@ -32,5 +32,23 @@ public NamedTypeSymbol ComImportCoClass
...
 
         #endregion
+
+        #region SkipLocalsInitAttribute
+        private bool _hasSkipLocalsInitAttribute;
+        public bool HasSkipLocalsInitAttribute
+        {
+            get
+            {
+                VerifySealed(expected: true);
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Portable/Symbols/Attributes/WellKnownAttributeData/TypeWellKnownAttributeData.cs]: >VerifySealed(expected: true); [](start = 16, length = 29)

While you are in this file, consider adding similar line into the getter above (for ComImportCoClass).

**Pull request merged**